### DISCUSSION TITLE
RAG does not recognize documents (when openwebui and ollama hostet on different servers)

### DISCUSSION THREAD

**Discussion by PierreSchwang** [open-webui/open-webui#3130] [Q&A] [answered]
Hi!

When using OpenWebUI the model does not seem to have any context whatsoever when sending the request to Ollama.
![grafik](https://github.com/open-webui/open-webui/assets/27054324/2e19e5f2-8145-49b4-8d17-3c167b7bcb2f)

The request to `/ollama/api/chat` contains the document reference:
```json
{
   "model":"llama3:latest",
   "messages":[
      {
         "role":"user",
         "content":"What is the document about?"
      }
   ],
   "options":{
      
   },
   "docs":[
      {
         "type":"doc",
         "collection_name":"d561508c36619dde4008325d703d2aef05d098489f90938220741b75fa798b5",
         "name":"zertifizierungsbericht-zer1780-xxx",
         "title":"Zertifizierungsbericht ZER-1780 xxx.pdf",
         "filename":"Zertifizierungsbericht ZER-1780 xxx.pdf",
         "content":{
            
         },
         "user_id":"aad8bfcb-fa3e-4732-a5fc-0ee2d4ee6ea3",
         "timestamp":1718285572,
         "upload_status":true
      }
   ],
   "citations":true,
   "chat_id":"f4ba9990-c5cd-439e-a468-2e392a221a44"
}
```

The log does not indicate any error:
```
INFO:     127.0.0.1:58600 - "GET /api/v1/documents/ HTTP/1.0" 200 OK
INFO:apps.rag.main:file.content_type: application/pdf
INFO:apps.rag.main:store_data_in_vector_db [Document(page_content='xxx', metadata={'source': '/app/backend/data/uploads/Zertifizierungsbericht ZER-1780 xxx.pdf', 'page': 0, 'start_index': 0}), Document(page_content='xxxx', metadata={'source': '/app/backend/data/uploads/Zertifizierungsbericht ZER-1780 xxx.pdf', 'page': 1, 'start_index': 0}), Document(page_content='xxx', metadata={'source': '/app/backend/data/uploads/Zertifizierungsbericht ZER-1780 xxx.pdf', 'page': 1, 'start_index': 1374}), Document(page_content='xxx', metadata={'source': '/app/backend/data/uploads/Zertifizierungsbericht ZER-1780 xxx.pdf', 'page': 2, 'start_index': 0})]
---AND MANY MORE TOKENIZED PARTS OF THE PDF---
d561508c36619dde4008325d703d2aef05d098489f90938220741b75fa798b5
Batches: 100%|██████████| 1/1 [00:00<00:00,  7.54it/s]
INFO:     10.91.255.121:0 - "POST /rag/api/v1/doc HTTP/1.1" 200 OK
INFO:     127.0.0.1:58624 - "POST /api/v1/documents/create HTTP/1.0" 200 OK
INFO:     127.0.0.1:58640 - "GET /api/v1/documents/ HTTP/1.0" 200 OK
INFO:     127.0.0.1:41278 - "GET /api/v1/users/user/settings HTTP/1.0" 200 OK
INFO:     127.0.0.1:41284 - "GET /api/v1/users/user/settings HTTP/1.0" 200 OK
INFO:     127.0.0.1:35680 - "POST /api/v1/chats/new HTTP/1.0" 200 OK
INFO:     127.0.0.1:35688 - "GET /api/v1/chats/ HTTP/1.0" 200 OK
Received "usage" event from zdBO0b-q9uCKQmzsAAA5: {'action': 'chat', 'model': 'llama3:latest', 'chat_id': 'f4ba9990-c5cd-439e-a468-2e392a221a44'}
Models in use: ['llama3:latest']
remove_after_timeout zdBO0b-q9uCKQmzsAAA5 llama3:latest
Received "usage" event from zdBO0b-q9uCKQmzsAAA5: {'action': 'chat', 'model': 'llama3:latest', 'chat_id': 'f4ba9990-c5cd-439e-a468-2e392a221a44'}
Models in use: ['llama3:latest']
remove_after_timeout zdBO0b-q9uCKQmzsAAA5 llama3:latest
Received "usage" event from zdBO0b-q9uCKQmzsAAA5: {'action': 'chat', 'model': 'llama3:latest', 'chat_id': 'f4ba9990-c5cd-439e-a468-2e392a221a44'}
Models in use: ['llama3:latest']
remove_after_timeout zdBO0b-q9uCKQmzsAAA5 llama3:latest
Received "usage" event from zdBO0b-q9uCKQmzsAAA5: {'action': 'chat', 'model': 'llama3:latest', 'chat_id': 'f4ba9990-c5cd-439e-a468-2e392a221a44'}
Models in use: ['llama3:latest']
remove_after_timeout zdBO0b-q9uCKQmzsAAA5 llama3:latest
Received "usage" event from zdBO0b-q9uCKQmzsAAA5: {'action': 'chat', 'model': 'llama3:latest', 'chat_id': 'f4ba9990-c5cd-439e-a468-2e392a221a44'}
Models in use: ['llama3:latest']
remove_after_timeout zdBO0b-q9uCKQmzsAAA5 llama3:latest
llama3:latest
INFO:     10.91.255.121:0 - "POST /api/chat/completed HTTP/1.1" 200 OK
INFO:     127.0.0.1:35702 - "POST /api/v1/chats/f4ba9990-c5cd-439e-a468-2e392a221a44 HTTP/1.0" 200 OK
INFO:     127.0.0.1:35704 - "GET /api/v1/chats/ HTTP/1.0" 200 OK
INFO:     127.0.0.1:35706 - "POST /api/v1/chats/f4ba9990-c5cd-439e-a468-2e392a221a44 HTTP/1.0" 200 OK
INFO:     127.0.0.1:35718 - "GET /api/v1/chats/ HTTP/1.0" 200 OK
INFO:     127.0.0.1:35734 - "GET /api/v1/chats/ HTTP/1.0" 200 OK
```

Ollama is not logging anything either.

The Ollama app and OpenWeb-UI container are running on two different servers in the same LAN. I can't even see any log output in OpenWeb-UI which would indicate a request to the Ollama tbh - but not sure if there should be any in the end.

**Comment by iraklishubitidze** [comment]
One addition:
We looked through the current issues and discussion and found [Discussion 2848](https://github.com/open-webui/open-webui/discussions/2848), which seems to be closest to the behaviour we are experiencing. We followed the presented solutions in the discussion including:

- Deleted all documents.
- Reset document settings back to default embedding model engine.
- Saved settings.
- Reset vector storage.
- Add the document back.
- Attempt to query the model about the document.

Sadly those didn´t work and the behaviour didn´t change in any notable way.
We also checked the ollama server for errors, but the logs didn´t show anything that cloud be related to this issue.

**Comment by tristan-mcinnis** [comment]
Did this get fixed?

**Reply by PierreSchwang** [reply]
> Did this get fixed?

The issue - sadly - still persists for us after updating (even onto the dev branch)

**Reply by iraklishubitidze** [reply, marked as answer]
So we actually solved it last week and as it seems it was a proxy error, relating to nginx.
1. We use very restricted servers, therefore we are in need to set a bunch of proxys before anything actually works (docker proxy, nginx, application proxys)
2. On an older version of OpenWebUI (before they implemented proxys as enviroment variables for docker) we had a deliberate proxy setup with nginx, which was routing all outgoing requests to the "right" adresses. This was when the issue first appeared.
3. After setting up a seperate environment (ollama + webui on one server) to compare the requests etc., we minified everything again and realized, that our proxy setup was unncessary since they implemented proxy support as a feature as docker environment variables.
4. We minified the nginx setup and all requests were correctly passed, therefore it worked again.
5. Sadly I can´t tell what caused the error in the first place, before proxy environment variables were implemented...

I hope this helps in any way. We won´t spend more time investigating the error for the older version, focusing on the working system now. :)
